### PR TITLE
Add Filter Tag Command

### LINKED ISSUE #224: Implement find by tag in tagging system

### PR DESCRIPTION
**Pull request by bryanljx**
Closes #224 

### Changes made
- Add abstract class for FilterCommand
- Add FilterTagCommand and FilterTagCommandParser
- Update CommandType, Messages, FoodRemParser accordingly
- Add test cases for FilterTagCommand, FilterTagCommandParser
- Update test cases for FoodRemParser, HelpCommand, HelpCommandParser

### CODE REVIEW

**Comment by Ferusel** [comment]
I suspect this will have merge conflicts after #286

**Comment by RichDom2185** [comment]
> I suspect this will have merge conflicts after #286

Yup, the GUI still has some ways to go, so we'll need to merge that first and block this, unfortunately.

**Review by Ferusel — APPROVED** [review]
LGTM!

File: src/main/java/seedu/foodrem/commons/enums/CommandType.java [hunk 1]
```diff
@@ -190,6 +190,19 @@ public String getUsage() {
...
+            return getCommandWord() + ": Filters all items in FoodRem by the specified tag"
+                    + "\n\n"
+                    + "Format:\n"
+                    + getCommandWord() + " "
+                    + PREFIX_NAME + "NAME\n\n"
+                    + "Example:\n"
+                    + getCommandWord() + " "
+                    + PREFIX_NAME + "Potatoes ";
+        }
+    },
```
Comment (RichDom2185) [inline src/main/java/seedu/foodrem/commons/enums/CommandType.java:205]: Can we insert this in the appropriate location following alphabetical order, like the rest?

**Review by RichDom2185 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/foodrem/commons/enums/CommandType.java (hunk 1 above)
Comment (RichDom2185) [reply src/main/java/seedu/foodrem/commons/enums/CommandType.java:205]: Oops, this was merged already.

**Pull request merged**